### PR TITLE
refactor(models/solver): harden solver runtime option boundaries (PR-1)

### PR DESCRIPTION
**Pull request by w5851**
## Summary
- Harden `Solver.jl` entry boundary by centralizing mode-agnostic runtime option parsing and validation into focused helpers (`xi/p_num/t_num/residual_norm_max` plus seed normalization), while keeping public `solve/solve_multi/solve_constraint` signatures unchanged.
- Keep orchestration flow explicit (`parse -> dispatch -> wrap`) for FixedMu and non-FixedMu paths, reducing inline branching and preparing clean handoff surface for PR-2 `ProblemSpec` slimming.
- Add regression-focused unit coverage for invalid runtime option types and keep behavior parity evidence (`fixedmu_use_problem_spec`) with targeted unit and integration smoke runs.

## Why
- This is PR-1 boundary hardening in the solver decoupling roadmap: reduce entry-layer coupling and hidden parameter drift risk before `ProblemSpec` and mode-kernel restructuring.

## Behavior-Parity Evidence
- `julia --project=. -e 'ENV["UNIT_FILES"]="models/test_solver.jl,models/test_solver_named_vec_parity.jl,models/test_solver_schema_adapter.jl"; include("tests/unit/runtests.jl")'`  
  - Result: `Unit | Pass 64 / Total 64`
- `julia --project=. -e 'ENV["INTEGRATION_PROFILE"]="smoke"; include("tests/integration/runtests.jl")'`  
  - Result: `Integration | Pass 463 / Total 463`

## Risks and Rollback
- Risk: runtime default/type migration could alter edge-case behavior.
- Mitigation: explicit argument validation + parity assertions in tests.
- Rollback: revert this PR as a whole to restore pre-hardening entry parsing behavior.

## PR-2 Handoff Notes
- `Solver.jl` runtime option and seed validation are now single-source helpers; PR-2 should avoid reintroducing entry-layer parsing logic and focus on internal `ProblemSpec` execution/diagnostic separation.

### CODE REVIEW

File: src/models/solver/Solver.jl [hunk 1]
```diff
@@ -93,25 +134,64 @@ end
...
         selector=selector,
+        seed_strategy=seed_strategy,
         seed_guess=seed_guess,
         seed_candidates=seed_candidates,
     )
 end
 
+@inline function _resolve_fixedmu_runtime_options(mode::FixedMu, T_fm::Real, μ_fm::Real, kwargs)
+    common = _resolve_common_runtime_options(kwargs)
+    auto_multiseed_fallback = Bool(get(kwargs, :auto_multiseed_fallback, true))
```
[the remark this answers is not in the document]
Comment (w5851) [reply src/models/solver/Solver.jl]: Fixed by replacing `Bool(get(...))` with explicit bool option validation via `_resolve_bool_option`, so invalid `auto_multiseed_fallback` now raises `ArgumentError` consistently with other runtime options.

File: src/models/solver/Solver.jl [hunk 2]
```diff
@@ -93,25 +134,64 @@ end
...
+
+@inline function _resolve_fixedmu_multi_runtime_options(mode::FixedMu, T_fm::Real, μ_fm::Real, kwargs)
+    common = _resolve_common_runtime_options(kwargs)
+    evaluate_all_attempts = Bool(get(kwargs, :evaluate_all_attempts, true))
+    seed_strategy = get(kwargs, :seed_strategy, MultiSeed())
+    explicit_seeds = if haskey(kwargs, :seeds)
+        _normalize_seed_pool(kwargs[:seeds], :seeds)
+    else
+        Vector{Vector{Float64}}()
+    end
```
[the remark this answers is not in the document]
Comment (w5851) [reply src/models/solver/Solver.jl:189]: Fixed in the same change: `_resolve_fixedmu_multi_runtime_options` and non-FixedMu `solve_multi` now use `_resolve_bool_option` for `evaluate_all_attempts`, with consistent `ArgumentError` on invalid types. Added unit coverage for both FixedMu and non-FixedMu invalid bool inputs.
